app/controllers/api/v2/play_roles_controller.rb

```diff
@@ -0,0 +1,93 @@
+module Api
+  module V2
+    # API controller for Ansible Plays
+    class PlayRolesController < ::Api::V2::BaseController
+      include ::Api::Version2
+      include ForemanTasks::Triggers
+
+      rescue_from ActiveRecord::RecordNotFound, :with => :resource_not_found
+
+      before_action :find_hosts, :only => [:play_roles_on_host]
+      before_action :find_hostgroups, :only => [:play_roles_on_hostgroup]
+
+      api :POST, '/ansible/play_roles_on_host', N_('Plays Ansible roles on hosts')
+      param :play_roles, Hash, :required => true do
+        param :host_ids, Array, :required => false, :desc => N_('Array of host IDs')
+        param :host_names, Array, :required => false, :desc => N_('Array of host names')
+      end
+
+      def play_roles_on_host
+        @result = []
+
+        @hosts.uniq.each do |host|
+          @result.append(
+            :host => host, :foreman_tasks => async_task(
+              ::Actions::ForemanAnsible::PlayHostRoles, host
+            )
+          )
+        end
+
+        render :json => @result
+      end
+
+      api :POST, '/ansible/play_roles_on_hostgroup', N_('Plays Ansible roles on hostgroups')
+      param :play_roles, Hash, :required => true do
+        param :host_ids, Array, :required => false, :desc => N_('Array of host IDs')
+        param :host_names, Array, :required => false, :desc => N_('Array of host names')
+      end
+
+      def play_roles_on_hostgroup
+        @result = []
+
+        @hostgroups.uniq.each do |hostgroup|
+          @result.append(
+            :hostgroup => hostgroup, :foreman_tasks => async_task(
+              ::Actions::ForemanAnsible::PlayHostgroupRoles, hostgroup
+            )
+          )
+        end
+
+        render :json => @result
```

Comment:
You might want to `process_response @result` instead, to provide some feedback if making the tasks went wrong